### PR TITLE
livebook: fix service not starting when LIVEBOOK_COOKIE was not set

### LINKED ISSUE #329869: livebook: doesn't work as a service
### Describe the bug
`livebook` doesn't seem to work as a module

### Steps To Reproduce
1. follow livebook service [example](https://nixos.org/manual/nixos/stable/#module-services-livebook-basic-usage)
2. enter any env variables or use environment file with any variables
3. `nixos-rebuild switch`

first of all, its systemd unit does not appear to start by default. so i did `systemctl --user daemon-reload` and `systemctl --user start livebook.service`

but it doesn't work, i can't access it either by IP or by domain, which is set up in my nginx configuration as a reverse proxy. some strange things happen while the unit is active [1st screenshot]. also there's nothing in the journal except info messages about the unit itself
as i see it, livebook fails to start, but the service is active because *something* is going on with `/dev/urandom` inside this `..livebook-wrapped-wrapped` wrapper script

i tried to make it work several different ways: from using `environment` option instead of my environment file to writing my very own systemd unit in NixOS configuration file, also using home-manager like so:
```nix
  # ...
  systemd.user.services = {
    livebook = {
      Install = { WantedBy = [ "default.target" ]; };

      Service = {
        ExecStart = "${pkgs.livebook}/bin/livebook start";
        KillMode = "mixed";
        Restart = "always";
        EnvironmentFile = "/var/lib/livebook.env";
      };
    };
  };
  # ...
```

but that doesn't work too. surely i tried starting livebook manually - `LIVEBOOK_PORT=3333 LIVEBOOK_PASSWORD=pretty_password LIVEBOOK_DATA_PATH=/var/www/livebook livebook start`

and it works as expected [2nd screenshot]

### Expected behavior
1. livebook module/service works and is accessible
2. custom systemd livebook unit works and is accessible

### Screenshots

| screen 1 |
|----------------------------------------------------------------------------------------------|
| screen 2 |

### Additional context
NixOS 24.05.3214.575f3027caa1 (Uakari) x86_64 running on a VPS

### Notify maintainers

cc @Munksgaard 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.41, NixOS, 24.05 (Uakari), 24.05.3214.575f3027caa1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixos-24.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by scvalex**
## Description of changes

Fixes #329869 

The underlying problem is described in https://github.com/livebook-dev/livebook/issues/2691

Systemd disables SIGPIPE by default.  When `LIVEBOOK_COOKIE` is not set, livebook will attempt to generate one by running something like `cat /dev/urandom | env LC_ALL=C tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1`.  This pipeline should terminate when `head` reads one line and closes its input, but if SIGPIPE is disabled, the pipeline never terminates and hangs the livebook startup.

The solution is to add `IgnoreSIGPIPE = false;` to the service config.

This was not caught by the tests because we were setting `LIVEBOOK_COOKIE` there.  Without that, the tests replicate the problem.

In addition to the tests, I am also now running this change locally and livebook starts without `LIVEBOOK_COOKIE` as expected.  Calling `Node.get_cookie()` in a notebook results in `3tK68jkA0dIg6NfTiGYrXb877GkoLhf4` which implies that the random generation is indeed working.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by scvalex** [comment]
@Munksgaard Could you review this PR, please?

**Comment by Munksgaard** [comment]
It works! Thanks a lot.

Before we merge, may I suggest that you add a separate commit  to put either you, me or both of us as maintainers of the module? It should be [quite simple](https://nixos.org/manual/nixos/unstable/#sec-meta-attributes).

**Review by Munksgaard — APPROVED** [review]
Looks good, thanks!

@NixOS/beam Can any of you merge?

**Pull request merged**
